src/renderers/renderer.builder.ts

```diff
@@ -23,14 +28,19 @@ export class RendererBuilder {
           process.exit(1)
         }
       case 'mobile':
-        // TODO: return WikimediaMobile renderer
-        break
+        if (hasWikimediaMobileRestApi) {
+          return new WikimediaMobileRenderer()
+        }
+        logger.error('No available mobile renderer.')
+        process.exit(1)
       case 'auto':
         if (hasWikimediaDesktopRestApi) {
```

Comment:
Logic seems wrong, mobile rendering should have the top priority?!

Reply:
This brings us to the recent conversation about `auto` mode, where we have to choose `desktop` if no `forceRender` is specified and no `mobile` capability is present. Check this line of code [here](https://github.com/openzim/mwoffliner/blob/251b6bb93f737b107ae6f356c0870cac9f6036f4/src/util/saveArticles.ts#L257) where I implemented an `auto` mode as a fallback for `mobile`.